streams/shared/src/main/scala/zio/stream/experimental/ZSink.scala

```diff
@@ -154,6 +154,28 @@ class ZSink[-R, -InErr, -In, +OutErr, +L, +Z](val channel: ZChannel[R, InErr, Ch
   ): ZSink[R1, InErr1, In1, OutErr1, L, Z1] =
     contramapChunksM(f).mapM(g)
 
+  def filterInput[In1 <: In](p: In1 => Boolean): ZSink[R, InErr, In1, OutErr, L, Z] = {
+    lazy val loop: ZChannel[R, InErr, Chunk[In1], Any, InErr, Chunk[In], Any] =
+      ZChannel.readWith[R, InErr, Chunk[In1], Any, InErr, Chunk[In], Any](
```

Comment:
These can be simplified to `contramap(_.filter(p ))` :-) (same for filterInputM)

Reply:
Almost had it but found foldChunks first and tried with that but it was not good for this :)
Thanks!